### PR TITLE
fix: export types and add consistency to type naming

### PR DESCRIPTION
**Pull request by chrisolsen**

### CODE REVIEW

File: libs/react-components/src/lib/skeleton/skeleton.tsx [hunk 1]
```diff
@@ -14,13 +14,16 @@ export type SkeletonType =
   | "profile"
   | "article";
 
-type SkeletonSize = 1 | 2 | 3 | 4;
+export type GoASkeletonSize = 1 | 2 | 3 | 4;
+
+// legacy naming
+export type SkeletonType = GoASkeletonType;
```
[the remark this answers is not in the document]
Comment (chrisolsen) [reply libs/react-components/src/lib/skeleton/skeleton.tsx:20]: It wasn't initially exported

File: libs/react-components/src/lib/footer/footer.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/footer/footer.tsx:1]: You missed the interface here `export interface GoAFooterProps`

File: libs/react-components/src/lib/form-step/form-step.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/form-step/form-step.tsx:1]: Missed the interface here, `export interface GoAFormStepProps`

File: libs/react-components/src/lib/footer-meta-section/footer-meta-section.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/footer-meta-section/footer-meta-section.tsx:1]: Missed the interface here `export interface GoAFooterMetaSectionProps`

File: libs/react-components/src/lib/footer-nav-section/footer-nav-section.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/footer-nav-section/footer-nav-section.tsx:1]: Missed the interface here, and the export, `export interface GoAFooterNavSectionProps`

File: libs/react-components/src/lib/form-stepper/form-stepper.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/form-stepper/form-stepper.tsx:1]: Missed the interface here, `export interface GoAFormStepperProps`

File: libs/react-components/src/lib/page-block/page-block.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/page-block/page-block.tsx:1]: Forgot the interface here, `export interface GoAPageBlockProps`

File: libs/react-components/src/lib/pages/pages.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/pages/pages.tsx:1]: Forgot the interface here and the export, `export interface GoAPagesProps`

File: libs/react-components/src/lib/side-menu-group/side-menu-group.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/side-menu-group/side-menu-group.tsx:1]: Forgot the interface and the export, `export interface GoASideMenuGroupProps`

File: libs/react-components/src/lib/side-menu-heading/side-menu-heading.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/side-menu-heading/side-menu-heading.tsx:1]: Forgot the interface and the export, `export interface `GoASideMenuSectionTitleProps`

File: libs/react-components/src/lib/side-menu/side-menu.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/side-menu/side-menu.tsx:1]: Forgot the interface, `export interface GoASideMenuProps`

File: libs/react-components/src/lib/skeleton/skeleton.tsx [hunk 2]
```diff
@@ -34,9 +36,9 @@ declare global {
 
 export interface SkeletonProps extends Margins {
```
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/skeleton/skeleton.tsx:37]: `GoASkeletonProps`?

File: libs/react-components/src/lib/spinner/spinner.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/spinner/spinner.tsx:1]: Forgot the interface, `export interface GoASpinnerProps`

File: libs/react-components/src/lib/tab/tab.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/tab/tab.tsx:1]: Forgot the interface, `export interface GoATabItemProps`

File: libs/react-components/src/lib/table/table.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/table/table.tsx:1]: Forgot the interface, `export interface GoATableProps`

File: libs/react-components/src/lib/tabs/tabs.tsx
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/tabs/tabs.tsx:1]: Forgot the interface and the export, `export interface GoATabsProps`

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**